src/main/java/anto/TaskList.java

```diff
@@ -0,0 +1,218 @@
+                        (Event) newTask).from.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm")), (
+                        (Event) newTask).to.format(DateTimeFormatter.ofPattern("dd-MM-yyyy HH:mm"))));
+            }
+            writer.close();
+        } catch (IOException e) {
+            throw new AntoException("OOPS!!! IOException");
+        }
+    }
+
+    /**
+     * Returns current array list of tasks.
+     *
+     * @return ArrayList of tasks.
+     */
+    public ArrayList<Task> getStorage() {
+        return this.taskList;
+    }
+
+    /**
+     * Returns current number of tasks.
+     *
+     * @return Current number of tasks.
+     */
+    public int getLength() {
+        return this.taskList.size();
+    }
+
+    /**
+     * Marks task at index as done.
+     *
+     * @param index Index of task to mark as done.
+     * @throws AntoException Throws AntoException if there is an IO Exception.
+     */
+    public void markTaskAsDone(int index) throws AntoException {
+        String relativePath = "data/anto.txt";
+        Path absolutePath = Paths.get(relativePath).toAbsolutePath();
+        File antoFile = absolutePath.toFile();
+
+        String tempRelativePath = "data/tempFile.txt";
+        Path tempAbsolutePath = Paths.get(tempRelativePath).toAbsolutePath();
+        File tempFile = tempAbsolutePath.toFile();
+
+        try {
+            Scanner sc = new Scanner(antoFile);
+            FileWriter writer = new FileWriter(tempFile);
+
+            int line = 0;
+
+            while (sc.hasNextLine()) {
+                String currLine = sc.nextLine();
+                if (line == index) {
+                    writer.write(currLine.replace("| 0 |", "| 1 |")
+                            + System.getProperty("line.separator"));
+                } else {
+                    writer.write(currLine + System.getProperty("line.separator"));
+                }
+                line++;
+            }
+
+            writer.close();
+            sc.close();
+            tempFile.renameTo(antoFile);
+        } catch (IOException e) {
+            throw new AntoException("OOPS!!! IOException");
+        }
+
+        this.taskList.get(index).markAsDone();
+    }
+
+    /**
+     * Unmark task at index.
+     *
+     * @param index Index of task to unmark.
+     * @throws AntoException Throws AntoException if there is an IO Exception.
+     */
+    public void unmarkTask(int index) throws AntoException {
+        String relativePath = "data/anto.txt";
+        Path absolutePath = Paths.get(relativePath).toAbsolutePath();
+        File antoFile = absolutePath.toFile();
+
+        String tempRelativePath = "data/tempFile.txt";
+        Path tempAbsolutePath = Paths.get(tempRelativePath).toAbsolutePath();
+        File tempFile = tempAbsolutePath.toFile();
+
+        try {
+            Scanner sc = new Scanner(antoFile);
```

Comment:
Should all this logic be placed in Storage class instead? Since it is handling reading and writing to storage.